ksqldb-functional-tests/src/test/resources/historical_plans/json_sr_-_ARRAY_-_key_-_inference/6.2.0_1607115574018/spec.json

```diff
@@ -87,7 +87,7 @@
       "valueSchema" : {
         "type" : "object",
         "properties" : {
-          "COL1" : {
```

Comment:
This plan and the others related to `key - inference` do not apply to JsonSchemaConverter anymore. The QTT were failing when comparing the JSON-SR schemas. However, I tried running the same test in Prod and things work fine. 

I created a JSON-SR schema with a `COL1 INTEGER` field:
Schema
```
{
  "properties": {
    "COL1": {
      "description": "The integer type is used for integral numbers.",
      "type": "integer"
    }
  },
  "type": "object"
}
```

I then created a stream with a `FOO INTEGER` using the above schema, then insert/select from the stream which it worked:
```
ksql> create stream json(foo INT) with (kafka_topic='json', value_format='JSON_SR');

 Message        
----------------
 Stream created 
----------------

ksql> insert into json(foo) values(1);

ksql> select * from json;
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|FOO                                                                                                                                                                                     |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|1                                                                                                                                                                                       |
Query Completed
```

I noticed the SR schema changed now to after inserting the values:
```
{
  "properties": {
    "FOO": {
      "connect.index": 0,
      "oneOf": [
        {
          "type": "null"
        },
        {
          "connect.type": "int32",
          "type": "integer"
        }
      ]
    }
  },
  "type": "object"
}
```

Perhaps the above changes (renaming the field) is compatible with JSON_SR only? I tried with Protobuf/Avro schemas and they do not allow the QTT test case to pass. 

I modified the QTT plans to make QTT to work because Proto/Avro dot now allow such test and JSON_SR allowed me to do it in Production. I don't think there will be an incompatibility problem.

Reply:
Does this after this change, insertion on existing json_sr won't work because of incompatible schema?

Reply:
Synced offline, why `col1` was there previously was a mystery but `FOO` is correct now. @spena also test in prod and there's no issue of `col1`. So ignore this...